apps/aecuckoo/Makefile

```diff
@@ -1,101 +1,92 @@
 EXECUTABLES = \
-	priv/bin/mean30s-generic \
-	priv/bin/mean28s-generic \
-	priv/bin/mean16s-generic \
-	priv/bin/mean30s-avx2 \
-	priv/bin/mean28s-avx2 \
-	priv/bin/mean16s-avx2 \
-	priv/bin/lean30 \
-	priv/bin/lean28 \
-	priv/bin/lean16 \
-	priv/bin/verify30 \
-	priv/bin/verify28 \
-	priv/bin/verify16
-
-LIBBLAKE = priv/lib/libblake2b.so
-
-SRC_EXECUTABLES = $(patsubst %,c_src/src/%,$(foreach ex,$(patsubst priv/bin/%,%,$(EXECUTABLES)),$(SRC_EXECUTABLE_NOTDIR_$(ex))))
-
-# Distinguish by self-explanatory name the mean miner binaries
-# requiring AVX2 instructions.
-SRC_EXECUTABLE_NOTDIR_mean30s-generic = mean30sx1
-SRC_EXECUTABLE_NOTDIR_mean28s-generic = mean28sx1
-SRC_EXECUTABLE_NOTDIR_mean16s-generic = mean16sx1
-SRC_EXECUTABLE_NOTDIR_mean30s-avx2    = mean30sx8
-SRC_EXECUTABLE_NOTDIR_mean28s-avx2    = mean28sx8
-SRC_EXECUTABLE_NOTDIR_mean16s-avx2    = mean16sx8
-SRC_EXECUTABLE_NOTDIR_lean30          = lean30
-SRC_EXECUTABLE_NOTDIR_lean28          = lean28
-SRC_EXECUTABLE_NOTDIR_lean16          = lean16
-SRC_EXECUTABLE_NOTDIR_verify30        = verify30
-SRC_EXECUTABLE_NOTDIR_verify28        = verify28
-SRC_EXECUTABLE_NOTDIR_verify16        = verify16
-
-SRC_LIBBLAKE = $(patsubst priv/lib/%,c_src/src/%,$(LIBBLAKE))
-
-SRC_ARCH_FLAGS = -m64
+	mean29-generic \
+	mean29-avx2 \
+	lean29-generic \
+	lean29-avx2 \
+	mean15-generic \
+	lean15-generic
+
+PRIVEXECS = $(addprefix $(PRIV)/, $(EXECUTABLES))
+
+PRIV = priv/bin
+CUCKOO = c_src/src/cuckoo
+
+HDRS=$(CUCKOO)/cuckoo.h $(CUCKOO)/../crypto/siphash.h
+
+# Flags from upstream makefile
+OPT ?= -O3
+
+GPP_ARCH_FLAGS ?= -m64 -x c++ -march=native
```

Comment:
I am not sure about `-march=native` option.

Reply:
Usage of `-march=native` causes all CPU operations available to the machine that is compiling to be used in the compiled binary hence making the resulting binary potentially not able to be supported on other machines. This seems reverting f49528b0 . Could you please clarify.

----

This PR:
* Builds package passing `-m64 ... -march=native`- see [CI](https://circleci.com/gh/aeternity/epoch/40936):
```
(cd c_src/src/cuckoo; g++ -m64 -x c++ -march=native -std=c++11 -Wall -Wno-format -Wno-deprecated-declarations -D_POSIX_C_SOURCE=200112L -O3 -DPREFETCH -I.  -pthread -o mean29-generic -DSAVEEDGES -DNSIPHASH=1 -DEDGEBITS=29 mean.cpp ../crypto/blake2b-ref.c)
```

Master corresponding to this PR:
* Builds package passing only `-m64` (not `-march`) - see [CI](https://circleci.com/gh/aeternity/epoch/40713):
```
gcc -m64 -std=gnu11 -O -fPIC -shared -o libblake2b.so blake2b-ref.c
...
g++ -m64 -std=c++11 -Wall -Wno-format -Wno-deprecated-declarations -D_POSIX_C_SOURCE=200112L -O3 -DPREFETCH -I.  -pthread -o mean30sx1 -DSAVEEDGES -DNSIPHASH=1 -DEDGEBITS=29 mean_miner.cpp -L. -lblake2b
```

Reply:
Wasn't careful enough when merging the two Makefiles...